application/models/services/UserManager.php

```diff
@@ -68,22 +68,38 @@ public function canAssignRole()
     }
 
     /**
-     * Returns true if the managing user can edit the target user
+     * Returns true if the managing user can edit the attribute of the target user
+     * Return true if target is same then managing (user can always update himself)
      * @return bool
      */
     public function canEdit()
     {
         if (empty($this->managingUser) || empty($this->targetUser)) {
             return false;
         }
-
-        return
-            Permission::model()->hasGlobalPermission('superadmin', 'read', $this->managingUser->id)
-            || $this->targetUser->uid == $this->managingUser->id
-            || (
-                Permission::model()->hasGlobalPermission('users', 'update', $this->managingUser->id)
-                && $this->targetUser->parent_id == $this->managingUser->id
-            );
+        /* user can update himself */
+        if ($this->managingUser == $this->targetUser) {
+            return true;
+        }
+        /* forcedsuperamdin (user #1) can always update all */
+        if (Permission::isForcedSuperAdmin($this->managingUser->id)) {
+            return true;
+        }
+        /* forcedsuperamdin can not be update (except by another forcedsuperamdin done before) */
+        if (Permission::isForcedSuperAdmin($this->targetUser->uid)) {
+            return false;
+        }
+        /* If target user is superamdin : managingUser must be allowed to create superadmin */
+        if (Permission::model()->hasGlobalPermission('superadmin', 'read', $this->targetUser->uid)) {
+            return Permission::model()->hasGlobalPermission('superadmin', 'create', $this->managingUser->id);
```

Comment:
Or check if is parent ?

Reply:
Both. Need to be allowed to create superadmin as well as be its parent.

PS: permission superadmin-create, does it exist?

Reply:
Yes : superadmin read and create

Before no superadmin can create new superadmin (only superadmin #1) it's a new feature in 4.X (lost in the changelog …) https://github.com/LimeSurvey/LimeSurvey/pull/810

Reply:
> Both. Need to be allowed to create superadmin as well as be its parent.
> 

```
        if (Permission::model()->hasGlobalPermission('superadmin', 'read', $this->uid)) {
            return Permission::model()->hasGlobalPermission('superadmin', 'create', $managerId)
                && $this->parent_id == $managerId;
        }

```